### PR TITLE
Add example that initializes and updates contract

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Add an example that shows how to initialise and update a smart contract.

## Changes

- Add example

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: examples/init-update-contract.rs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+impl std::fmt::Display for WeatherError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "invalid weather variant; expected \"rainy\" or \"sunny\", but got \"{}\"",
+            self.0
+        )
+    }
+}
```
Comment (abizjak) [inline examples/init-update-contract.rs:86]: You can just derive `Error` here.

```
#[derive(Debug, Error)]
#[error("invalid weather variant; expected \"rainy\" or \"sunny\", but got \"{0}\"")
struct WeatherError(String);
```

File: examples/init-update-contract.rs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                param,
+            };
+
+            send::init_contract(
+                &keys.account_keys,
+                keys.address,
+                nonce,
+                expiry,
+                payload,
+                Energy { energy: 10000 },
```
Comment (abizjak) [inline examples/init-update-contract.rs:146]:
```suggestion
                10000u64.into()
```

File: examples/init-update-contract.rs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+                message,
+            };
+
+            send::update_contract(
+                &keys.account_keys,
+                keys.address,
+                nonce,
+                expiry,
+                payload,
+                Energy { energy: 10000 },
```
Comment (abizjak) [inline examples/init-update-contract.rs:164]:
```suggestion
                10000u64.into()
```

**Review by abizjak — APPROVED** [review]
Some minor code style suggestions. Looks good otherwise.

File: examples/init-update-contract.rs (hunk 1 above)
Comment (Bargsteen) [reply examples/init-update-contract.rs]: Ah, nice 👍

**Pull request merged**